### ISSUE TITLE
Zooming breaks with a large amount of data

### ISSUE THREAD

**Issue by amit1911** [highcharts/highcharts#6926]
#### Expected behaviour
Zoom in / Zoom out should work as expected with

#### Live demo with steps to reproduce
http://jsfiddle.net/amit1911/4Lh2odt0/1/
1. Drilldown Fruits.
2. Zoom in on any part. Everything is fine.
3. Zoom out. Now you will some empty space in the area where you zoomed in.
![zoom out bug](https://user-images.githubusercontent.com/3509404/27903080-9b8cb940-626a-11e7-93e3-b36518d9ab9b.gif)

#### Affected browser(s)
All

**Comment by amit1911** [comment]
This bug started appearing after 4.2.7 release. 4.2.6 is fine. Take a look at http://jsfiddle.net/amit1911/4Lh2odt0/5/.

**Comment by amit1911** [comment]
Upon further investigation I have found out that [https://github.com/highcharts/highcharts/commit/56e528076f78979464d65e8ea07e71ce0f4a18d2](url) commit causes this issue

**Comment by sebastianbochan** [comment]
Hi @amit1911 
    Thank you for the reporting. We will investigate it.

**Comment by pawelfus** [comment]
Thanks for reporting and the investigation. We really appreciate this! :)

**Comment by amit1911** [comment]
the `updateNames` function is the one causing this bug. If I disable it, then it works fine.

**Comment by amit1911** [comment]
Any updates on this? Need a fix urgently.

**Comment by pawelfus** [comment]
**Workaround:**

Set [`series.cropThreshold`](http://api.highcharts.com/highcharts/plotOptions.series.cropThreshold), for example: http://jsfiddle.net/4Lh2odt0/11/

It looks like related to or a duplicate of #5857

**Comment by amit1911** [comment]
thanks a lot...seems to be working...

**Comment by amit1911** [comment]
i do not have any maximum limit on the data so is it okay to set a cropThreshold to a very large number like 10000000 or is there any performance impact?

**Comment by pawelfus** [comment]
It's okay to set to a very large number, but if you will try to render 10 000 000 columns.. well, I don't recommend that :)

**Comment by amit1911** [comment]
Okay... What is the maximum number of points that highcharts can handle? I think we sometimes have around 100,000 points in our application. Can highcharts handle these many points easily?

**Comment by pawelfus** [comment]
Just to confirm: you have 100,000 of points with drilldown and categories on a one chart?

In that case I suggest to use boost.js module. Rendering 100,000 of SVG elements in DOM is not the best idea. For example chart's width=1000px, that would require 100 bars in 1px.

**Comment by amit1911** [comment]
Yeah, i think this is the worst case scenario. We have a lot of data. This is why I use zooming so that users can interact with individual points as we have sorting and drilldown as well. Btw what is the maximum data (estimate) thatÏ highcharts can handle without lag.

**Comment by pawelfus** [comment]
It really depends on a browser, machine (mobile vs PC), other features on the site etc. I can not say any reliable value.

Just to be on the same page: bug is related only to `axis.type ='category'` and categories' names in points. Using `axis.categories` and points without category name will work fine without `cropThreshold`: http://jsfiddle.net/4Lh2odt0/12/ - I think this solution is more suitable for your case and will not reduce the performance.

**Comment by amit1911** [comment]
Okay.. I'll give this a try and see the performance difference. But should I still use boost module? Is there any downside to using it?

**Comment by pawelfus** [comment]
With 100,000 I would still consider using boost. Boost is now in beta, so not all options are available. Limits are listed inside the file: http://code.highcharts.com/modules/boost.src.js

**Comment by amit1911** [comment]
Okay I'll give it a try as well. Thanks a lot for all your help...i was really stuck with this issue.
